pydantic/types.py

```diff
@@ -626,3 +634,238 @@ def display(self) -> str:
 
     def get_secret_value(self) -> bytes:
         return self._secret_value
+
+
+class Color:
+    __slots__ = '_original', '_color_match'
+
+    def __init__(self, value: ColorType) -> None:
+        self._original: ColorType = value
+        self._color_match: Optional[str] = None
+        self._parse_color()
+
+    @staticmethod
+    def _rgb_str_to_tuple(value: str) -> Optional[RGBAType]:
+        """
+        Return RGB/RGBA tuple from the passed string.
+
+        If RGBA cannot be matched, return None
+        """
+        r = re.compile(
+            r'rgba?\((?P<red>\d{1,3}),\s*'
+            r'(?P<green>\d{1,3}),\s*'
+            r'(?P<blue>\d{1,3})'
+            r'(,\s*(?P<alpha>\d{1}\.\d{1,}))?\)',
+            re.IGNORECASE,
+        )
+        match = r.match(value)
+
+        if match is None:
+            return None
+
+        red = int(match.group('red'))
+        green = int(match.group('green'))
+        blue = int(match.group('blue'))
+        alpha = float(match.group('alpha')) if match.group('alpha') is not None else 1.0
+        return red, green, blue, alpha
+
+    @staticmethod
+    def _almost_equal(value_1: float, value_2: float = 1.0) -> bool:
+        """
+        Return True if two floats are almost equal
+        """
+        return abs(value_1 - value_2) <= 1e-8
+
+    @staticmethod
+    def _tuple_to_rgb(value: Optional[Tuple[Any, ...]]) -> Optional[RGBType]:
+        """
+        Convert a tuple to RGB tuple, if it fails raise an error
+        """
+
+        def is_int_color(c: Any) -> bool:
```

Comment:
please move this out of the function, best to static method, or just a plain method.